File: subtract-the-product-and-sum-of-digits-of-an-integer.kt
class Solution {
  fun subtractProductAndSum(n: Int): Int {
    val string = n.toString()
    val numbers = string.map { it.toString().toInt() }
    val product = numbers.reduce { acc, it ->  acc * it }
    val sum = numbers.reduce { acc, it ->  acc + it }
    return product - sum
  }
}

File: subtract-the-product-and-sum-of-digits-of-an-integer.rs
impl Solution {
    pub fn subtract_product_and_sum(n: i32) -> i32 {
        let str = n.to_string();
        let mut v: Vec<i32> = vec![];
        let mut product = 1;
        let mut sum = 0;
        for c in str.chars() {
            v.push(c.to_digit(10).unwrap() as i32);
        }
        for n in v {
            product *= n;
            sum += n;
        }
        return product - sum;
    }
}
